sdk/keyvault/Azure.Security.KeyVault.Shared/KeyVaultIdentifier.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+
+namespace Azure.Security.KeyVault
+{
+    internal struct KeyVaultIdentifier
```

Comment:
Should this be used in all projects while you're at it?

Reply:
added issue #7845 to refactor how Ids are parsed holistically